datajoint/expression.py

```diff
@@ -661,10 +661,10 @@ def cursor(self, offset=0, limit=None, order_by=None, as_dict=False):
         """
         if offset and limit is None:
             raise DataJointError("limit is required when offset is set")
-        self._top = dict(
-            offset=offset,
-            limit=limit,
-            order_by=order_by,
+        self._top = Top(
```

Comment:
No, this does not belong here. These are the sorting values from the fetch, they should not override the existing `.top` proprty.

Reply:
The top level `QueryExpression` will never have a `_top` property since `Tops` immediately terminate into a subquery.

Reply:
This may need to be fixed because now it sounds like `dj.Top` always generates a subquery even when one is unecessary.

Since now you have this new mechanism for `& dj.Top`, you can re-use inside `fetch` and consolidate the code rather than handling it separately here.